### PR TITLE
Move to SSP UI

### PR DESCRIPTION
**Pull request by cicnavi**
Move OIDC module pages into standard SSP UI. 

This PR starts with extending base SSP template for OIDC module pages. Also, some new admin pages are introduced, like overview of module config values related to OIDC and federation settings. The main horizontal menu is taken from `admin` module.

image

### CODE REVIEW

**Review by tvdijen — COMMENTED** [review]
Looking good, @cicnavi !!

I think the `Menu.php`, `Menu\Item.php` and `menu.twig` are eligible for porting to SSP-core, so they can be reused by other modules.

**Comment by cicnavi** [comment]
@tvdijen I know there are a lot of changes in this PR, but if you would take a look again, maybe you catch something...

File: public/assets/css/src/default.css [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+.oidc-container {
+    display: flex;
+    max-width: inherit;
+    margin: 0 auto;
+}
+
+/* Style for the left menu */
+.menu {
+    min-width: 200px;
+    /*background-color: #f4f4f4;*/
```
Comment (tvdijen) [inline public/assets/css/src/default.css:33]: Maybe remove unused CSS?

File: src/Admin/Authorization.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        return $this->sspBridge->utils()->auth()->isAdmin();
+    }
+
+    /**
+     * @throws \SimpleSAML\Module\oidc\Exceptions\AuthorizationException
+     */
+    public function requireAdmin(bool $forceAdminAuthentication = false): void
+    {
+        if ($forceAdminAuthentication) {
```
Comment (tvdijen) [inline src/Admin/Authorization.php:31]: `=== true` strict comparison.
Maybe this is less relevant with a `bool` typed param, but it's a defense-in-depth

File: src/Controllers/Admin/ClientController.php [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+    /**
+     * @throws \SimpleSAML\Error\ConfigurationError
+     * @throws \SimpleSAML\Error\Exception
+     * @throws \SimpleSAML\Module\oidc\Exceptions\OidcException
+     */
+    public function add(): Response
+    {
+        $form = $this->formFactory->build(ClientForm::class);
+
+        if ($form->isSuccess()) {
```
Comment (tvdijen) [inline src/Controllers/Admin/ClientController.php:165]: strict comparison

**Review by tvdijen — COMMENTED** [review]
Really neat job @cicnavi ! I would not have done this any other way!
I've limited my review to the PHP src-files (left out tests and twig-templates)

Not related to this PR, but in general:
- You may want to consider adding PHP 8.4 to the testsuite now that it has been released last month.
- Also, I'm in the process of migrating all repositories from Psalm to PHPStan.

**Comment by cicnavi** [comment]
Great, thanks!
Ok, will add PHP 8.4 check in another PR. 
I've added a TODO note about moving to PHPStan in upgrade doc, so will do at some point.

**Pull request merged**